engine/src/main/java/com/datatorrent/stram/LaunchContainerRunnable.java

```diff
@@ -253,6 +253,7 @@ public void run()
     vargs.add(String.format("-D%scid=%s", StreamingApplication.DT_PREFIX, jvmID));
     vargs.add("-Dhadoop.root.logger=" + (dag.isDebug() ? "DEBUG" : "INFO") + ",RFA");
     vargs.add("-Dhadoop.log.dir=" + ApplicationConstants.LOG_DIR_EXPANSION_VAR);
+    vargs.add(String.format("-D%s='%s'", LogicalPlan.APPLICATION_NAME.getLongName(), dag.getValue(LogicalPlan.APPLICATION_NAME)));
```

Comment:
Fixed.

Per man bash:

Enclosing characters in single quotes preserves the literal value of each character within the quotes. A single quote may not occur between single quotes, even when preceded by a backslash.